app/main.py

```diff
@@ -1,2 +1,68 @@
+from __future__ import annotations
+from typing import Any, Union, Tuple
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.threshold = int(self.capacity * 2 / 3)
+        self.resize_multiplier = 2
+        self.size = 0
+        self.hash_table = [[]] * self.capacity
+
+    def __len__(self) -> int:
+        return self.size
+
+    def _get_index_and_hash(
+            self,
+            key: Union[bool, int, float, str, tuple]
+    ) -> Tuple[int, int]:
+        index = hash(key) % self.capacity
+        hash_of_item = hash(key)
+        return hash_of_item, index
+
+    def __setitem__(
+            self,
+            key: Union[bool, int, float, str, tuple],
```

Comment:
```suggestion
            key: Hashable,
```
None type, functions, classes and so on also can be key.